### DISCUSSION TITLE
> And I tested the latest OpenWRT image (`openwrt-bcm2711-rpi-4-ext4-rtl8168-factory.img.gz`, flashed with Etcher), and it booted up and let me add the WAN port / eth1 without issue:

### DISCUSSION THREAD

**Discussion by David-McGuire** [geerlingguy/raspberry-pi-pcie-devices#453] [General]
> And I tested the latest OpenWRT image (`openwrt-bcm2711-rpi-4-ext4-rtl8168-factory.img.gz`, flashed with Etcher), and it booted up and let me add the WAN port / eth1 without issue:
> 
> <img alt="Screen Shot 2022-01-21 at 10 04 42 AM" width="1280" src="https://user-images.githubusercontent.com/481677/150560013-001a980a-0c1a-4225-9eed-e4c650ca3179.png">
> 
> Now if the software config for that OLED display gets sorted, I think this board will have everything fixed up out of the box except the PoE issue. I saw a comment about pulling resistors earlier, but it looks like that comment was removed. @yoyojacky are the hardware engineers still taking a look at that? Thanks!

Hi bro, 
Thanks for your testing, that's cool, and our products has been removed the resistor on board which cause the PoE issue. the product on stock is good to go.

_Originally posted by @yoyojacky in https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/337#issuecomment-1155924019_

**Comment by David-McGuire** [comment]
Help needed installing the openwrt on the Raspberry Pi CM4 Router Carrier Board by 52Pi

So ive tried to use the (openwrt-22.03.0-rc4-bcm27xx-bcm2711-rpi-4-ext4-factory.img) to load this and It will not load.

Steps I have taken to install openwrt

1. downloaded the (openwrt-22.03.0-rc4-bcm27xx-bcm2711-rpi-4-ext4-factory.img) from (https://downloads.openwrt.org/snapshots/targets/bcm27xx/bcm2711/)
2. decompiled this file
3. Flashed to SD card using BalenEtcher
4. Inserted the SD card

 to [Raspberry Pi CM4 Router Carrier Board by 52Pi]
5. Enabled by jumper caps, to Device mode and Host mode both and same results.
6. Powered up 

Steps taken to install Raspberry Pi OS

1. I have also tried to install the Raspberry Pi OS using the steps n  (https://wiki.52pi.com/index.php/EP-0146#Operating_System) with the same results.

![20220628_162443](https://user-images.githubusercontent.com/82978221/176278504-4cd05bd3-e4d7-46e5-b5e1-e333f4811b78.jpg)

Any help here to get (openwrt) installed I would greatly appreciate.

**Comment by yoyojacky** [comment]
SD:　card not detected... please check the TF card and try again?

**Reply by David-McGuire** [reply]
I reformatted the  (4) different SD card's and completed to Flashed to SD card using BalenEtcher
Inserted the SD card
to [Raspberry Pi CM4 Router Carrier Board by 52Pi]

Enabled by jumper caps, to Device mode and Host mode both and same results.

Does any of the jumpers on the eMMC version CM4 burning switch need to be connected as at this point they are not?

Thoughts?

**Reply by PixlRainbow** [reply]
CM4s with onboard eMMC use the default SD card lanes to connect the eMMC, so the default SD pins aren't actually attached to anything on the eMMC models.
If the router board is not using GPIO 22-27 as auxiliary SD to connect the external SD card slot, then eMMC CM4s will not be able to read external SD cards.
Edit: The official documentation on seeedstudio's website does say the SD card slot can't be used on eMMC models; you will have to burn the onboard eMMC using a PC connected via USB.

**Reply by David-McGuire** [reply]
Sorry for the stupid Question but let me see if I understand this, Please correct me if I misunderstand this 
1. Move the jumper to the nRPIBOOT - Bootloader
2. Move the jumper to the Host for boot
3. Connect via the PC using USB C port on the Raspberry Pi CM4 Router Carrier Board
4. Using Etcher Flash the "openwrt-bcm27xx-bcm2711-rpi-4-ext4-with_oled096_factory.img
5. Connect a LAN cable to the Port 0
6. Remove jumper to the nRPIBOOT - Bootloader
7. Remove 
8. Plugged in HDMI Cable
9. Removed PC unplugging the USB C port on the Raspberry Pi CM4 Router Carrier Board 
10. Plugged in Power via USB C
11. See the boot loader
12. Now i can see the boot loading but it does not complete

Not sure what the next step is as it is stuck at 26.018946 IPV: ADDCONFIG )NETDEV_CHANGE: br-lan: link becomes ready?

**Reply by PixlRainbow** [reply]
That is the boot complete. It doesn't have a GUI. Try visiting web interface via it's IP using another device's web browser.

**Reply by David-McGuire** [reply]
How would i find the IP of the device? just look at the current IPs on the network?

**Reply by PixlRainbow** [reply]
That's one way. The other way is that since it is a router, you can plug a device into it's downstream Ethernet port and then just visit the gateway IP (e.g. 192.168.1.1)

**Reply by David-McGuire** [reply]
thanks Ill check this out

**Reply by David-McGuire** [reply]
Thanks everyone I got this set up :)

**Reply by David-McGuire** [reply]
The only other question I have is the OLED I can seem to get it working, I saw posts about this and I installed the version with the OLED but I get the following (*/etc/int.d/oled emable) do i need to add the files in the "luci-app-oled.zip" file to this? Also I have the WAN port connected to the Router/I currently have and I can seem to connect to the Internet (Port0), from the PC I have on the LAN port (Port1). I can access the Webpage but not outside the router.

**Reply by PixlRainbow** [reply]
Have you visited Services > OLED page and ticked "Enable"?
If you visit Network > Interfaces page do you see a "WAN" interface displayed there?
Also, your message appears to be malformed:
> but I get the following (*/etc/int.d/oled emable) do i need to add the files in the "luci-app-oled.zip" file to this?  

Did you mean to put something after "I get the following"?

**Reply by David-McGuire** [reply]
for the OLED I have tried that and no luck

When O boot eth1 is not shown 

Created eth1 and these are the results 
 
As for the eth1 (Wan I have tried a few different things
Connected laptop to Eth0. able to get in the gui webpage and make the following
Eth1 connected to laptop (using ip of 192.168.1.194
Eth0 connected to ISP ..

Network design ... Modem -> router on eth1
Network design with laptop -> modem 
1. DHCP and not pulling an IP address from the internet provider Deleted this interface and re added it.
 2. Connected laptop to modem verified ip information as follows.
   IP Address : xx.x.214.133
   Subnet Mask: 255.255.248.0
   Gateway: xx.x.240.1
   DHCP Server: xx.x.193.1
   DNS: xx.x.0.184
 b. Changed interface on Eth1 to static and set to the same address that I get when laptop is connected (Spoke to ISP and they confirmed that there is a lease time of 120 days but that this also would be the same address when it gets renewed).
   IP Address : xx.x.214.133
   Subnet Mask: 255.255.248.0
   Gateway: xx.x.240.1
   DHCP Server: xx.x.193.1
   DNS: xx.x.0.184

When the Modem is directly plugged into the router I get RX and TX but no IP address, 
Any thoughts
